### PR TITLE
remove usage of ohc parent pom

### PR DESCRIPTION
**Pull request by maggu2810**
The openHAB Distro repo is the only one I know about that is still using a POM of openHAB Core "feature/p2/poms" directory (or subdirectory).

Please check if all used stuff has been moved to this POM (I have tried it but I don't know if I succeed).

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,16 +1,8 @@
...
-    <parent>
-        <groupId>org.openhab</groupId>
-        <artifactId>pom</artifactId>
-        <version>2.5.0-SNAPSHOT</version>
-        <relativePath/>
-    </parent>
-
     <groupId>org.openhab.distro</groupId>
     <artifactId>pom</artifactId>
-    <!-- do not delete the version here as it is required by the release process -->
```
Comment (kaikreuzer) [inline pom.xml]: Why did you remove this notice?

File: pom.xml [hunk 2]
```diff
@@ -185,7 +180,7 @@
                     <plugin>
                         <groupId>org.eclipse.tycho</groupId>
                         <artifactId>tycho-versions-plugin</artifactId>
-                        <version>${tycho.version}</version>
+                        <version>1.4.0</version>
```
Comment (kaikreuzer) [inline pom.xml]: If no tycho stuff is left, we probably also won't need the tycho-versions-plugin anymire - I guess the "prepare-release" profile can be removed (it was used for the old manual release process).

File: pom.xml (hunk 1 above)
Comment (maggu2810) [reply pom.xml]: Dropping the version of a POM makes only sense if you use a parent one.
If you don't use a parent POM, the version will not be removed at all.
So no need to state it explicit.

Or can you tell me the version of that coordinate if no parent is used and no one is set. :wink:

File: pom.xml (hunk 2 above)
Comment (maggu2810) [reply pom.xml]: The execution ID `update-version` has been configured to execute the `set-version` goal.
So, I assume that function is used anywhere.

File: pom.xml (hunk 1 above)
Comment (kaikreuzer) [reply pom.xml]: Good point 🙄

File: pom.xml (hunk 2 above)
Comment (kaikreuzer) [reply pom.xml]: I doubt it, but ok, no need to remove it in this PR.

**Comment by wborn** [comment]
Can you fix the [build error](https://ci.openhab.org/job/PR-openHAB-Distribution/89/console)?

`[ERROR] Failed to execute goal org.apache.maven.plugins:maven-install-plugin:3.0.0-M1:install (default-install) on project distro: NoFileAssignedException: The packaging plugin for this project did not assign a main file to the project but it has attachments. Change packaging to 'pom'. -> [Help 1]`

We ran into it before and fixed it by pinning the maven-install-plugin to 2.5.2. 

See:
* https://github.com/openhab/openhab-core/pull/408
* [MINSTALL-151](https://issues.apache.org/jira/browse/MINSTALL-151)

**Comment by maggu2810** [comment]
Sure, done.
Let's wait and check the build.
